### PR TITLE
Upgrade to Play 2.5 and rewrite half the app in the process

### LINKED ISSUE #301: Upgrade to play 2.5
We should modernise Riff-Raff to play 2.5.

During this process we should change the CSRF implementation introduced in #300 to use the global filter with some holes punched in it for the API.

### LINKED ISSUE #325: Remove IRC support
@sihil There's no reason to keep this around, right? We don't have an IRC server these days?

### PR DESCRIPTION
**Pull request by cb372**
It nearly killed me, but I have upgraded the Play app to 2.5 and introduced compile-time DI.

Fixes #301 

Also:

* Remove support for the `deployinfo.json` way of looking up resources, as it has been superseded by Prism
* Remove IRC support (fixes #325)
* Use a filter for CSRF instead of wrapping actions with `CSRFCheck`
* Upgrade Scanamo because its cats dependency conflicted with that of the latest play-googleauth
* Update a few dependencies
* Fix all deprecation warnings
* Fix a bunch of IntelliJ warnings, code formatting, general tidy up
* Fix to use `-CODE` Dynamo table when running locally

### CODE REVIEW

File: riff-raff/app/controllers/api.scala [hunk 1]
```diff
@@ -214,15 +208,15 @@ object Api extends Controller with Logging with LoginActions {
       case default => 5
     }
 
-    val deploys = deploysByState.map { case (state, deployList) =>
-      val seriesDataByDay = deployList.groupBy(_.time.toLocalDate).mapValues(_.size).toList.sortBy {
-        case (date, _) => date
+    val deploys = deploysByState.map { case (state, deploysInThatState) =>
+      val seriesDataByDay = deploysInThatState.groupBy(_.time.toLocalDate).mapValues(_.size).toList.sortBy {
```
Comment (philwills) [inline riff-raff/app/controllers/api.scala]: 👍

File: riff-raff/app/controllers/api.scala [hunk 2]
```diff
@@ -214,15 +208,15 @@ object Api extends Controller with Logging with LoginActions {
...
+        case (day, _) => day
       }
       val seriesJson = Graph.zeroFillDays(seriesDataByDay, firstDate, lastDate).map {
-        case (day, deploys) =>
+        case (day, deploysOnThatDay) =>
```
Comment (philwills) [inline riff-raff/app/controllers/api.scala]: 👍

**Comment by philwills** [comment]
👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 

This is EPIC!

Lots of lovely little improvements aside from the main Play upgrade.

'javascripts' does set my teeth on edge though.

**Comment by cb372** [comment]
Thanks for the review!

There are a few more things I wanted to do but didn't get around to. I've created issues for them.

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.3.7")
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.5.6")
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-web" % "1.1.1")
-addSbtPlugin("com.typesafe.sbt" % "sbt-less" % "1.0.4")
 addSbtPlugin("com.typesafe.sbt" % "sbt-coffeescript" % "1.0.0")
-addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.1.1")
+addSbtPlugin("com.typesafe.sbt" % "sbt-less" % "1.1.0")
+//addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.1.1")
```
Comment (sihil) [inline project/plugins.sbt]: Kill this comment?

**Comment by sihil** [comment]
This is fantastic. Nice to see the CSRF filter replacing the individual checks.

It's starting to look like a maintainable project now 👍  :shipit:

**Comment by adamnfish** [comment]
Is the compile-time DI migration missing the `LoggerConfigurator` stuff, or am I missing something?

https://www.playframework.com/documentation/2.5.x/ScalaCompileTimeDependencyInjection#Application-entry-point

**Pull request merged**